File: Difficulty: Easy/Count numbers/count-numbers.cpp
class Solution {
  public:
    int count(int N) {
        int c = 0;
        
        for (int i = 1;i<=N;i++){
            bool isthree = false;
            int temp = i;
            
            while(temp!=0){
                int digit = 0;
                digit = temp%10;
                if(digit == 3){
                    isthree = true;
                    break;
                }
                temp = temp/10;
                
            }
            if(!isthree)
            c++;
        }
        
        return c;
        
    }
};

File: Difficulty: Easy/Count numbers/count-numbers.py
class Solution:
    def count(self, N):
        c = 0
        
        for i in range(1, N + 1):
            temp = i
            isthree = False
            
            while temp != 0:
                digit = temp % 10
                
                if digit == 3:
                    isthree = True
                    break
                
                temp = temp // 10   # ✅ FIX HERE
            
            if not isthree:
                c += 1
                
        return c